engines/immortal/kernal.cpp

```diff
@@ -0,0 +1,1107 @@
+		if (index >= 0x8000) {								// If negative, it's a row to draw
+			// rowY is (I think) the position of the start of the scroll window within the tile data
+			rowY = (index & 0x7FFF) + _myUnivPointY;
+
+			// The background is a matrix of rows and columns, so for each column, we draw each row tile
+			for (int i = 0; (i < (kViewPortCW + 1)); i++) {
+				//draw the column of rows
+				while (_columnIndex[i] < ((kViewPortCW + 1) * (kViewPortCH + 1))) {
+
+					uint16 k = _myModLCNM[i][_columnIndex[i]];
+					// ******* This is just so that the array can be indexed right now, will remove when myModLCNM is actually useable
+					k = 0;
+					// *****************************
+					if ((rowY - kChrDy[k]) < _columnTop[i]) {
+						break;
+					}
+					if (kIsBackground[k] == 0) {
+						// If it's a background tile, we already drew it (why is it in here then??)
+						if (kChrMask[k] >= 0x8000) {
+							// Right Mask, draw lower right hand corner (LRHC)
+							drawLRHC(_myCNM[i][_columnIndex[i]], _columnTop[i], _columnX[i]);
+
+						} else if (kChrMask[k] == 0) {
+							// Floor or cover, draw the whole CHR
+							drawSolid(_myCNM[i][_columnIndex[i]], _columnTop[i], _columnX[i]);
+
+						} else {
+							// Left Mask, draw lower left hand corner (LLHC)
+							drawLLHC(_myCNM[i][_columnIndex[i]], _columnTop[i], _columnX[i]);
+						}
+					}
+					_columnTop[i] += kChrH;
+					_columnIndex[i] += (kViewPortCW + 1);
+				}
+			}
+
+		} else {
+			// If positive, it's a sprite
+			uint16 x = (_sprites[index]._X - _myViewPortX) + kVSX;
+			uint16 y = (_sprites[index]._Y - _myViewPortY) + kVSY;
+			superSprite(_sprites[index]._dSprite, x, y, _sprites[index]._image, kVSBMW, _screenBuff, kMySuperTop, kMySuperBottom);
+		}
+		n++;
+	} while (n != _num2DrawItems);
+}
+
+void ImmortalEngine::backspace() {
+	// Just moves the drawing position back by a char, and then draws an empty rect there
+	_penX -= 8;
+	//rect(_penX + 32, 40, 8, 16, 0);
+
+	// The Y is hardcoded here presumably because it's only used for the certificate
+	for (int y = 0; y < 16; y++) {
+		for (int x = 0; x < 8; x++) {
+			_screenBuff[((y + 40) * kResH) + (x + (_penX + 32))] = 0;
+		}
+	}
+}
+
+void ImmortalEngine::printByte(int b) {
+	int hundreds = 0;
+	int tens = 0;
+
+	while ((b - 100) >= 0) {
+		hundreds++;
+		b -= 100;
+	}
+
+	if (hundreds > 0) {
+		printChr(char (hundreds + '0'));
+	}
+
+	while ((b - 10) >= 0) {
+		tens++;
+		b -= 10;
+	}
+
+	if (tens > 0) {
+		printChr(char (tens + '0'));
+	}
+
+	printChr(char (b + '0'));
+}
+
+void ImmortalEngine::printChr(char c) {
+	// This draws a character from the font sprite table, indexed as an ascii char, using superSprite
+	c &= kMaskASCII;				// Grab just the non-extended ascii part
+
+	if (c == ' ') {
+		_penX += 8;					// A space just moves the position on the screen to draw ahead by the size of a space
+		return;
+	}
+
+	// Why is single quote done twice?
+	if (c == 0x27) {
+		_penX -= 2;
+	}
+
+	switch (c) {
+		case 'm':
```

Comment:
Incorrect indentation. Switch is not adding an indentation level per our guidelines.